test/custom-parser.3.test.js

```diff
@@ -58,13 +56,12 @@ test('contentTypeParser should add a custom parser with RegExp value', t => {
     })
   })
 
-  fastify.listen({ port: 0 }, err => {
-    t.error(err)
-
-    t.teardown(() => fastify.close())
+  fastify.listen({ port: 0 }, async err => {
+    t.assert.ifError(err)
 
-    t.test('in POST', t => {
+    await t.test('in POST', (t) => {
```

Comment:
we would still need to add in `done` and call after last assert to ensure the planned asserts were run
```suggestion
    await t.test('in POST', (t, done) => {
```

Reply:
Done! see here: [8ccbf71](https://github.com/fastify/fastify/pull/5903/commits/8ccbf71c4bfddae19cd2e07926a58ade1f610813)